datacube_stats/tasks.py

```diff
@@ -195,20 +195,19 @@ def set_task(self, task, filtered_times, extra_fn_args):
         """
         remove_index_list = list()
         for i, sr in enumerate(task.sources):
-            for k, v in sr.items():
-                if k == 'data':
-                    if self.filter_product.get('method') == "by_hydrological_months":
-                        all_dates = [s.strftime("%Y-%m-%d") for s in
-                                     v.sources.time.values.astype('M8[s]').astype('O').tolist()]
-                    elif self.filter_product.get('method') == "by_tide_height":
-                        all_dates = [s.strftime("%Y-%m-%dT%H:%M:%S") for s in
-                                     v.sources.time.values.astype('M8[s]').astype('O').tolist()]
-                    if bool(set(all_dates) & set(filtered_times)):
-                        v.sources = v.sources.isel(time=[i for i, item in enumerate(all_dates) if item in
-                                                         filtered_times])
-                        _LOG.info("source included %s", v.sources.time)
-                    else:
-                        remove_index_list.append(i)
+            v = sr.data
+            if self.filter_product.get('method') == "by_hydrological_months":
+                all_dates = [s.strftime("%Y-%m-%d") for s in
+                             v.sources.time.values.astype('M8[s]').astype('O').tolist()]
+            elif self.filter_product.get('method') == "by_tide_height":
+                all_dates = [s.strftime("%Y-%m-%dT%H:%M:%S") for s in
+                             v.sources.time.values.astype('M8[s]').astype('O').tolist()]
+            if bool(set(all_dates) & set(filtered_times)):
```

Comment:
I know it's not your code Imam... But someone should add a pylint check for this sort of redundancy. `if` will always convert something to a bool, it doesn't have to be specified like this.